index.bs

```diff
@@ -119,105 +133,462 @@ A device is [=capable of supporting=] the depth sensing feature if the device ex
+  readonly attribute XRDepthUsage depthUsage;
+  readonly attribute XRDepthDataFormat depthDataFormat;
+};
+</script>
+
+The {{XRSession/depthUsage}} describes depth sensing usage with which the session was configured. If this attribute is accessed on a session that does not have depth sensing enabled, the user agent MUST throw an {{InvalidStateError}}.
+
+The {{XRSession/depthDataFormat}} describes depth sensing data format with which the session was configured. If this attribute is accessed on a session that does not have depth sensing enabled, the user agent MUST throw an {{InvalidStateError}}.
+
 Obtaining depth data {#obtaining-data}
 ====================
 
 XRDepthInformation {#xr-depth-info-section}
 ------------------
 
 <script type="idl">
-
 [SecureContext, Exposed=Window]
 interface XRDepthInformation {
-  [SameObject] readonly attribute Uint16Array data;
-
   readonly attribute unsigned long width;
   readonly attribute unsigned long height;
 
   [SameObject] readonly attribute XRRigidTransform normTextureFromNormView;
-
-  float getDepth(unsigned long column, unsigned long row);
+  readonly attribute float rawValueToMeters;
 };
+</script>
+
+The {{XRDepthInformation/width}} attribute contains width of the [=XRDepthInformation/depth buffer=] (i.e. number of columns). When accessed, the algorithm to <a lt="attempting to access members">access a member</a> MUST be run.
+
+The {{XRDepthInformation/height}} attribute contains height of the [=XRDepthInformation/depth buffer=] (i.e. number of rows). When accessed, the algorithm to <a lt="attempting to access members">access a member</a> MUST be run.
+
+The {{XRDepthInformation/normTextureFromNormView}} attribute contains a {{XRRigidTransform}} that needs to be used when indexing into the [=XRDepthInformation/depth buffer=]. The transformation that the matrix represents changes the coordinate system from normalized view coordinates to normalized texture coordinates. When accessed, the algorithm to <a lt="attempting to access members">access a member</a> MUST be run.
+
+The {{XRDepthInformation/rawValueToMeters}} attribute contains conversion factor from raw values that could be obtained directly from the [=XRDepthInformation/depth buffer=] into meters. When accessed, the algorithm to <a lt="attempting to access members">access a member</a> MUST be run.
+
+Each {{XRDepthInformation}} has an associated <dfn for=XRDepthInformation>view</dfn> that stores {{XRView}} from which the depth information instance was created.
+
+Each {{XRDepthInformation}} has an associated <dfn for=XRDepthInformation>depth buffer</dfn> that contains depth buffer data. Different {{XRDepthInformation}}s may store objects of different concrete types in the depth buffer.
+
+<div class="algorithm" data-algorithm="accessing-depth-information-members">
+
+When <dfn>attempting to access members</dfn> of {{XRDepthInformation}} or any interface that inherits from it, the user agent MUST run the following steps:
```

Comment:
I'm sorry if I've overlooked this aspect of the API in previous reviews, but reading through it now I question if this check is necessary on every member access?

We return a new instance of the DepthInformation variant on each frame, right? Given that's the case there's no penalty to allowing the base class attributes to remain accessible after the depth buffer it contains has "expired". In fact, in may be useful to keep around a previous frame's instance just to compare and see if the width/height/etc. has changed between frames.

The important part is that the actual depth data becomes inaccessible outside of the frame, and for those members/methods the algorithm described is still appropriate.

Reply:
Correct, we definitely have to protect against accessing the depth buffer itself, we can make the other attributes accessible even outside of the frame if we think that this is useful. I mostly made them inaccessible for consistency - it felt a bit weird to have a "partially dead" object, but if that is not a concern, there should be no issue w/ relaxing this.

Reply:
ArrayBuffers already have a concept of "Neutering" to remove access to their underlying data (such as when transferred to a worker) and WebGL textures can be deleted to remove access to the underlying GPU resource while leaving the opaque object behind, so I don't think that it's particularly strange.

Reply:
Resolving, please reopen if you think the current changes are insufficient.